flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/SourceReaderBase.java

```diff
@@ -211,7 +226,12 @@ private boolean moveToNextSplit(
 
         currentSplitContext = splitStates.get(nextSplitId);
         checkState(currentSplitContext != null, "Have records for a split that was not registered");
-        currentSplitOutput = currentSplitContext.getOrCreateSplitOutput(output);
+        currentSplitOutput =
+                currentSplitContext.getOrCreateSplitOutput(
```

Comment:
Would it be simpler and more readable to pass a `Function<T, Boolean>`  instead of passing `eofRecordEvaluator`, `splitId` and `splitFetcherManager`? Then we won't need to add the extra generic type parameters to `SplitContext`.

I made a commit here [1] to show the resulting code.

[1] https://github.com/apache/flink/commit/215f80fe4640dc6686ca9ebddc6ae42c5fdb4cb3